system/src/system_network_internal.cpp

```diff
@@ -17,6 +17,62 @@
 
 #include "system_network_internal.h"
 
+namespace particle {
+
+namespace {
+
+bool networkOff(network_interface_index iface) {
+    if (network_ready(iface, NETWORK_READY_TYPE_ANY, nullptr)) {
+        network_off(iface, 0, 0, nullptr);
+        return true;
+    }
+    return false;
+}
+
+void networkOn(network_interface_index iface) {
+    network_on(iface, 0, 0, nullptr);
+}
+
+} // namespace
+
+void resetNetworkInterfaces() {
+    // TODO: There's no cross-platform API to enumerate available network interfaces
+#if HAL_PLATFORM_MESH
+    const bool resetMesh = networkOff(NETWORK_INTERFACE_MESH);
```

Comment:
Why not simply call `network_off()` with `NETWORK_INTERFACE_ALL` ?

Reply:
I didn't want to change the state of interfaces which are turned on but have no link. The system network API doesn't provide much visibility into an interface's state, unfortunately. On second thought, I think we should also invoke `network_connect()` for affected interfaces in this function.

Reply:
Ah, okay, makes sense. :+1: I hope we'll be able to resolve that in Q4. I have a WIP branch that should allow to manage/query individual states.